src/stores/RoomViewStore.js

```diff
@@ -176,6 +177,22 @@ class RoomViewStore extends Store {
                 this._joinRoom(payload);
             }
         } else if (payload.room_alias) {
+            // Try the room alias to room ID navigation cache first to avoid
+            // blocking room navigation on the homeserver.
+            const roomId = getCachedRoomIDForAlias(payload.room_alias);
```

Comment:
how about flipping this around to be

```suggestion
            let roomId = getCachedRoomIDForAlias(payload.room_alias);
            if (!roomId) {
                try {
                    const result = await MatrixClientPeg.get().getRoomIdForAlias(payload.room_alias);
                    storeRoomAliasInCache(payload.room_alias, result.room_id);
                    roomId = result.room_id;
                } catch (err) {...}
                
                dis.dispatch({...});
```

to not have two identical dispatches next to each other on diff branches

Reply:
Okay, seems fine to me. (Thought about it before, but couldn't decide what I preferred...)

Reply:
Okay, seems fine to me. (Thought about it before, but couldn't decide what I preferred...)